### PR TITLE
fix: fallback to twitter nft contract address

### PR DESCRIPTION
**Pull request by lelenei**
## Description

Closes # (NO_ISSUE)

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

**Review by Jack-Works — DISMISSED** [review]

**Review by Jack-Works — DISMISSED** [review]

File: packages/web3-shared/evm/types/index.ts [hunk 1]
```diff
@@ -655,6 +655,7 @@ export enum AddressNameType {
     RSS3 = 'RSS3',
     GUN = 'GUN',
     THE_GRAPH = 'THE_GRAPH',
+    TWITTER = 'TWITTER',
```
Comment (guanbinrui) [inline packages/web3-shared/evm/types/index.ts]: Plz rename it to `TWITTER_BLUE`.

File: packages/plugins/Debugger/src/SNSAdaptor/components/TabContent.tsx [hunk 2]
```diff
@@ -7,6 +7,7 @@ export interface TabContentProps {
 }
 
 export function TabContent({ identity, addressNames }: TabContentProps) {
+    console.log(identity)
```
Comment (guanbinrui) [inline packages/plugins/Debugger/src/SNSAdaptor/components/TabContent.tsx]:
```suggestion
```

File: packages/mask/src/social-network-adaptor/twitter.com/utils/user.ts [hunk 3]
```diff
@@ -35,11 +36,12 @@ export const getNickname = () => {
 export const getTwitterId = () => {
     const node = searchNickNameSelector().evaluate()?.parentElement?.parentElement?.firstChild?.nextSibling?.firstChild
         ?.firstChild?.lastChild as HTMLDivElement
-    if (!node) return ''
-
-    const twitterIdNode = node.querySelector('div span')
-    if (!twitterIdNode) return ''
-    return twitterIdNode.innerHTML.trim().replace('@', '')
+    if (node) {
```
Comment (guanbinrui) [inline packages/mask/src/social-network-adaptor/twitter.com/utils/user.ts:39]: Just code style changes?

File: packages/mask/src/components/InjectedComponents/ProfileTabContent.tsx [hunk 4]
```diff
@@ -46,11 +47,30 @@ export function ProfileTabContent(props: ProfileTabContentProps) {
     const [selectedTab, setSelectedTab] = useState<Plugin.SNSAdaptor.ProfileTab | undefined>()
 
     const identity = useCurrentVisitingIdentity()
-    const { value: addressNames, loading: loadingAddressNames } = useAddressNames(identity)
+    const { value: addressNames = [], loading: loadingAddressNames } = useAddressNames(identity)
 
+    const { loading: loadingTwitterAddressName, value: twitterAddressName } = useNFTContainerAtTwitter(
```
Comment (guanbinrui) [inline packages/mask/src/components/InjectedComponents/ProfileTabContent.tsx]: Not a separated hook. Plz make it integrate into `packages/mask/src/plugins/Wallet/services/domain.ts`.

File: packages/mask/src/social-network-adaptor/twitter.com/utils/user.ts [hunk 5]
```diff
@@ -53,15 +65,17 @@ export const getPersonalHomepage = () => {
 }
 
 export const getAvatar = () => {
-    const node = searchAvatarSelector().evaluate() as HTMLImageElement
-    if (!node) return ''
+    let node = searchAvatarSelector().evaluate()
+    if (!node) node = searchNFTAvatarSelector().evaluate()
```
Comment (guanbinrui) [inline packages/mask/src/social-network-adaptor/twitter.com/utils/user.ts]:
```suggestion
    const node = searchAvatarSelector().evaluate() || searchNFTAvatarSelector().evaluate()
```

File: packages/mask/src/social-network-adaptor/twitter.com/utils/user.ts [hunk 6]
```diff
@@ -35,11 +36,22 @@ export const getNickname = () => {
...
     const node = searchNickNameSelector().evaluate()?.parentElement?.parentElement?.firstChild?.nextSibling?.firstChild
         ?.firstChild?.lastChild as HTMLDivElement
-    if (!node) return ''
+    if (node) {
+        const twitterIdNode = node.querySelector('div span')
+        if (twitterIdNode) return twitterIdNode.innerHTML.trim().replace('@', '')
+    }
+
+    let ele = searchAvatarSelector().evaluate()?.closest('a')
+    if (!ele) ele = searchNFTAvatarSelector().evaluate()?.closest('a')
```
Comment (guanbinrui) [inline packages/mask/src/social-network-adaptor/twitter.com/utils/user.ts]:
```suggestion
    const ele = searchAvatarSelector().evaluate()?.closest('a') || searchNFTAvatarSelector().evaluate()?.closest('a')
```

**Review by guanbinrui — DISMISSED** [review]

**Review by guanbinrui — DISMISSED** [review]

**Comment by guanbinrui** [comment]
@DimensionDev/qa

**Comment by AinuX** [comment]
Need @lelenei to do further investigation.

**Comment by AinuX** [comment]
Checked

File: packages/mask/src/web3/context.ts [hunk 7]
```diff
@@ -28,6 +33,24 @@ import type { InternalSettings } from '../settings/createSettings'
...
+import CryptoPunks from '@masknet/web3-contracts/abis/CryptoPunks.json'
+
+const PUNK_CONTRACT_ADDRESS = '0xb47e3cd837ddf8e4c57f05d70ab865de6e193bbb'
+async function getERC721ContractOwner(address: string, tokenId: string): Promise<string> {
+    const web3 = createWeb3(Services.Ethereum.request, () => ({
+        chainId: ChainId.Mainnet,
+    }))
+    if (isSameAddress(address, PUNK_CONTRACT_ADDRESS)) {
+        const PUNKContract = createContract(web3, PUNK_CONTRACT_ADDRESS, CryptoPunks as AbiItem[])
+        return PUNKContract?.methods.punkIndexToAddress(tokenId).call() ?? ''
```
Comment (guanbinrui) [inline packages/mask/src/web3/context.ts]: Just allow to return of undefined.

File: packages/mask/src/web3/context.ts [hunk 8]
```diff
@@ -121,7 +144,25 @@ function createWeb3Context(disablePopup = false, isMask = false): Web3ProviderTy
         getAssetsList: WalletRPC.getAssetsList,
         getAssetsListNFT: WalletRPC.getAssetsListNFT,
         getCollectionsNFT: WalletRPC.getCollectionsNFT,
-        getAddressNamesList: WalletRPC.getAddressNames,
+        getAddressNamesList: async (identity: Parameters<typeof WalletRPC.getAddressNames>[0]) => {
+            const addressNames = await WalletRPC.getAddressNames(identity)
+            if (identity.identifier.network === 'twitter.com') {
+                const result = await UserNFTContainerAtTwitter.getUserNftContainer(identity.identifier.userId ?? '')
+                if (result && result.type_name.toUpperCase() === 'ERC721') {
```
Comment (guanbinrui) [inline packages/mask/src/web3/context.ts]: result?.type_name

File: packages/mask/src/web3/context.ts [hunk 9]
```diff
@@ -28,6 +33,24 @@ import type { InternalSettings } from '../settings/createSettings'
...
+async function getERC721ContractOwner(address: string, tokenId: string): Promise<string> {
+    const web3 = createWeb3(Services.Ethereum.request, () => ({
+        chainId: ChainId.Mainnet,
+    }))
+    if (isSameAddress(address, PUNK_CONTRACT_ADDRESS)) {
+        const PUNKContract = createContract(web3, PUNK_CONTRACT_ADDRESS, CryptoPunks as AbiItem[])
+        return PUNKContract?.methods.punkIndexToAddress(tokenId).call() ?? ''
+    }
+    const ERC721Contract = createContract<ERC721>(web3, address, ERC721ABI as AbiItem[])
+    return ERC721Contract?.methods.ownerOf(tokenId).call() ?? ''
```
Comment (guanbinrui) [inline packages/mask/src/web3/context.ts]: The same as above.

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**